scenarios/aws/vm/ec2vm/ec2VM.go

```diff
@@ -105,6 +81,39 @@ func newVM(ctx *pulumi.Context, options ...ec2params.Option) (*EC2VM, error) {
 	}, nil
 }
 
+type EC2UnixVM struct {
+	Infra
+	*commonvm.UnixVM
+}
+
+// NewUnixEc2VMWithEnv creates a new EC2 instance with an existing aws.NewEnvironment. By default use WithOS(os.UbuntuOS, os.AMD64Arch).
+// The returned vm provides additional methods compared to NewEc2VM
+func NewUnixEc2VMWithEnv(env aws.Environment, options ...ec2params.Option) (*EC2UnixVM, error) {
+	vm, err := NewEC2VMWithEnv(env, options...)
+	if err != nil {
+		return nil, err
+	}
+	unixVM, err := commonvm.NewUnixVM(vm.VM)
+	if err != nil {
+		return nil, err
+	}
+
+	return &EC2UnixVM{
+		UnixVM: unixVM,
+		Infra:  vm.Infra,
+	}, nil
+}
+
+// NewUnixEc2VMW creates a new EC2 instance with an existing aws.NewEnvironment. By default use WithOS(os.UbuntuOS, os.AMD64Arch).
```

Comment:
typo: incorrect comment copy/pasted from above

Reply:
The comment is actually wrong: `creates a new EC2 instance with an existing aws.NewEnvironment. By default use WithOS(os.UbuntuOS, os.AMD64Arch).`

While it does create the `aws.Environment` itself.